chunky/src/java/se/llbit/chunky/plugin/ChunkyPlugin.java

```diff
@@ -0,0 +1,112 @@
+/* Copyright (c) 2017 Chunky contributors
+ *
+ * This file is part of Chunky.
+ *
+ * Chunky is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Chunky is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License
+ * along with Chunky.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package se.llbit.chunky.plugin;
+
+import se.llbit.chunky.Plugin;
+import se.llbit.chunky.main.Version;
+import se.llbit.json.JsonObject;
+import se.llbit.json.JsonParser;
+import se.llbit.log.Log;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.nio.file.FileSystem;
+import java.nio.file.FileSystems;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * A chunky plugin.
+ */
+public class ChunkyPlugin {
+  private final JsonObject meta;
+  private final Plugin implementation;
+
+  protected ChunkyPlugin(JsonObject meta, Plugin implementation) {
+    this.meta = meta;
+    this.implementation = implementation;
+  }
+
+  /**
+   * Gets the name of this plugin.
+   *
+   * @return the name of this plugin
+   */
+  public String getName() {
+    return meta.get("name").stringValue("");
+  }
+
+  /**
+   * Gets the version of this plugin.
+   *
+   * @return the version of this plugin
+   */
+  public String getVersion() {
+    return meta.get("version").stringValue("<Unknown version>");
+  }
+
+  /**
+   * Gets the implementation of this plugin.
+   *
+   * @return the implementation of this plugin
+   */
+  public Plugin getImplementation() {
+    return implementation;
+  }
+
+  public static ChunkyPlugin load(File pluginJar) throws LoadPluginException {
+    Map<String, String> env = new HashMap<>();
+    env.put("create", "true");
```

Comment:
Indeed, the file will always exist here, so [no env is needed](http://docs.oracle.com/javase/7/docs/technotes/guides/io/fsp/zipfilesystemproviderprops.html).